### PR TITLE
Add recursive import support for sqlite3

### PR DESCRIPTION
**Pull request by jkowens**
As requested in #247.

Sqlite doesn't support concurrent write operations, so it can be assumed all ids for a batch insert will be sequential.

This also extracts the recursive import tests out into a shared example so that they can be applied to other adapters. If all adapters eventually get this functionality, the specs could be moved into the main spec file.

### CODE REVIEW

File: test/support/shared_examples/recursive_import.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        assert_difference "Topic.count", +1 do
+          Topic.import new_topics, recursive: true, on_duplicate_key_update: [:updated_at], validate: false
+          new_topics.each do |topic|
+            assert_not_nil topic.id
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (zdennis) [inline test/support/shared_examples/recursive_import.rb]: :thumbsup: Thanks for pulling these out.

**Comment by zdennis** [comment]
@jkowens, this looks good. It looks like AR 5.0 and AR 3.1 aren't happy. I'm less worried about 3.1 as we're going to deprecating that soon. 

Any thoughts on what's up with AR 5.0?

**Comment by jkowens** [comment]
Hey @zdennis, merging #280 will fix the failing AR 5.0 tests. The latest version of Seamless Database Pool is now compatible with AR 5.0 RC1 but it is no longer compatible with AR 3.1.

**Comment by zdennis** [comment]
Thanks @jkowens !

**Comment by zdennis** [comment]
This is going out in the 0.14.0 release.

**Pull request merged**